lib/ASRouter.jsm

```diff
@@ -380,6 +404,7 @@ class _ASRouter {
     await this.setState({messageBlockList, providerBlockList, messageImpressions, providerImpressions, previousSessionEnd});
     this._updateMessageProviders();
     await this.loadMessagesFromAllProviders();
+    await MessageLoaderUtils.cleanupCache(this.state.providers, storage);
```

Comment:
@k88hudson I added the cache cleanup at the end of `init()`. Thoughts?

Reply:
that seems reasonable yep!